test/kontrol/README.md

```diff
@@ -0,0 +1,200 @@
+# Euler Vault Kit Kontrol Tests
+-------------------------
+
+In this repo you'll find a series of examples showcasing some of the features of Kontrol, an open-source tool for formal verification of Solidity smart contracts. 
+
+## Getting Kontrol
+
+We recommend installing Kontrol via the K framework package manager, `kup`.
+
+To get `kup` and install Kontrol:
+```shell
+bash <(curl https://kframework.org/install)
+kup install kontrol
+```
+
+For more information you can visit the [`kup` repo](https://github.com/runtimeverification/kup) and the [`kup` cheatsheet](https://docs.runtimeverification.com/kontrol/cheatsheets/kup-cheatsheet).
+
+## Using Docker Instead
+
+We also provide a docker image with this repository cloned and all the commands already executed in case you want to walk through the instructions provided.
+
+To get the docker image and run a bash shell, please run:
+```shell
+docker run -it ghcr.io/runtimeverification/kontrol/friday-workshop
+```
+
+Once you've entered the container, please run the following command to use a Foundry profile we have set up for this tutorial:
+```shell
+export FOUNDRY_PROFILE=kontrol
+```
+
+To see the list of proofs that have been generated for this project, run:
+```shell
+kontrol list
+```
+
+## `run-kontrol.sh`
+
+It is customary with Kontrol projects to have a `run-kontrol.sh` script containing the steps to reproduce the proofs and some other features.
+In this example we have a top-level [`run-kontrol.sh`](../../run-kontrol.sh) which executes commands needed to run all proofs included in this workshop.
+
+### Kontrol usage
+
+[`test/kontrol/proofs`](./proofs) contains Solidity files with Kontrol tests (`*.k.sol` and `*.t.sol`) we are using in this tutorial. To build the proofs, run:
+```shell
+kontrol build
+```
+in this (`test/kontrol`) directory. The arguments this command is executed with are specified in the [`kontrol.toml`](../../kontrol.toml) file.
+
+To run the proofs, execute:
+```shell
+kontrol prove
+```
+
+A specific proof can be run by passing the `--mt` flag with the name of the proof. For example, to run the [`prove_Allowance_ReturnsZero`](./proofs/DTokenProof.k.sol) proof for the `DToken` contract, execute:
+```shell
+kontrol prove --mt prove_Allowance_ReturnsZero
+```
+This command will ensure that `DToken` contract's `allowance` function returns zero for _any_ input addresses.
+
+### Examples
+
+In this repository, you can find the following examples:
+
+#### 1. [Counter](./proofs/Counter.k.sol)
+
+Taken from the [Kontrol documentation](https://docs.runtimeverification.com/kontrol/guides/kontrol-example/property-verification-using-kontrol), `CounterTest` is a simple (but eloquent) example of how Kontrol can catch an issue missed by a fuzzer.
+Please note that, at the moment, this example is not available in the Docker container.
```

Comment:
Should we remove this and just re-create the docker image after this PR is merged?

Reply:
Yeah, that's an option too! I wasn't sure if we'll make it today (I'll need to be afk for a couple of hours) and I've been meaning to send a follow-up message with it sooner than later... maybe we can send it as is, and then I'll update the PR once the Docker is re-built?